src/transformers/utils/quantization_config.py

```diff
@@ -418,3 +428,53 @@ def post_init(self):
                     f"""dataset needs to be either a list of string or a value in
                     ['wikitext2','c4','c4-new','ptb','ptb-new'], but we found {self.dataset}"""
                 )
+        if self.bits == 4:
+            if self.disable_exllama is not None:
+                logger.warning(
+                    "Using `disable_exllama` is deprecated and will be removed in version 4.37. Use `use_exllama` instead and specify the version with `exllama_version`."
+                    "The value of `use_exllama` will be overwritten by `disable_exllama` passed in `GPTQConfig` or stored in your config file."
+                )
+                self.use_exllama = not self.disable_exllama
+
+            if self.use_exllama:
+                if self.exllama_version == ExllamaVersion.ONE:
+                    logger.warning(
+                        "You have activated exllama backend. Note that you can get better inference "
+                        "speed using exllamav2 kernel by setting `exllama_version=2`."
+                    )
+                if self.exllama_version == ExllamaVersion.TWO:
+                    optimum_version = version.parse(importlib.metadata.version("optimum"))
+                    autogptq_version = version.parse(importlib.metadata.version("auto_gptq"))
+                    if optimum_version <= version.parse("1.13.2") or autogptq_version <= version.parse("0.4.2"):
+                        raise ValueError(
+                            f"You need optimum > 1.13.2 and auto-gptq > 0.4.2 . Make sure to have that version installed - detected version : optimum {optimum_version} and autogptq {autogptq_version}"
+                        )
+
+    def to_dict_optimum(self):
+        """
+        Get compatible dict for optimum gptq config
+        """
+        quant_dict = self.to_dict()
+        # make it compatible with optimum config
+        quant_dict["disable_exllama"] = not (self.use_exllama and self.exllama_version == ExllamaVersion.ONE)
+        quant_dict["disable_exllamav2"] = not (self.use_exllama and self.exllama_version == ExllamaVersion.TWO)
```

Comment:
This is because in the PR that was merged to add exllamav2, we used `disable_exllamav2` in optimum. I will fix this in `optimum`, so that we use the same logic !